lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -560,7 +560,7 @@ export class AuthorizationCodeClient extends BaseClient {
                 let accountLoginHintClaim = this.extractLoginHint(request.account);
 
                 if (accountLoginHintClaim && request.domainHint) {
-                    this.logger.warning(`AuthorizationCodeClient.createAuthCodeUrlQueryString: "domainHint" param is set, skipping opaque "login_hint" claim`);
+                    this.logger.warning(`AuthorizationCodeClient.createAuthCodeUrlQueryString: "domainHint" param is set, skipping opaque "login_hint" claim. Please consider not passing domainHint`);
```

Comment:
Shouldn't this be `opaqueLoginHint` is preferred over `domain_hint`? 
Also I am a little concerned that a set of users who deem to *need* domain_hint will now keep seeing more interaction?

Reply:
Given that ESTS does not allow both "domain_hint" and opaque "login_hint" to be present I can think of 2 options here:
1. Skip opaque "login_hint" and set it to "account.username".
2. Remove "domain_hint" if opaque "login_hint" claim is present.

I would prefer the first one as per recommendation from ESTS.

Reply:
Ideally an opaque login_hint should be priority #1 and domain_hint is priority #4 (after sid and upn) but this gets a little weird in scenarios where the developer explicitly provides domain_hint but one of the other hints is provided implicitly via account (provided or via active account). Another less opinionated approach might be to mirror eSTS behavior and simply throw forcing the developer to be clearer about their intention. 

I'm on the fence about what I like better so I'm ok with whatever you decide.

Reply:
Isn't legacy login_hint `account.username` and the opaque login_hint, a hash of certain properties? In that case, isn't removing `domain_hint` the better option?

Can we get this clarity for STS?